p2p/src/interface/p2p_interface.rs

```diff
@@ -13,6 +13,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#[derive(Debug, serde::Serialize)]
+pub struct ConnectedPeer {
+    pub peer_id: String,
+    pub addr: String,
+}
```

Comment:
Is it possible to have more descriptive types that serialize to / deserialize from a `String`?

Reply:
Nice idea, thanks. But let's please wait until we replace libp2p. Peer IDs and addresses have different types in libp2p and mock. Once removed we should be able to use `MockPeerId` and some type for peer address directly.
I added TODO.